src/main/java/org/gelecekbilimde/scienceplatform/post/controller/CategoryController.java

```diff
@@ -25,16 +23,35 @@ class CategoryController {
 
 	private final CategoryToResponseMapper categoryToResponseMapper = CategoryToResponseMapper.initialize();
 
-	@GetMapping
+	@GetMapping("/categories")
 	SuccessResponse<List<CategoryResponse>> getCategoryList() {
-		List<Category> categories = categoryService.getCategories();
+		List<Category> categories = categoryService.findAll();
 		List<CategoryResponse> categoryResponses = categoryToResponseMapper.map(categories);
 		return SuccessResponse.success(categoryResponses);
 	}
 
-	@PostMapping("/create")
-	SuccessResponse<Void> createCategory(@RequestBody @Valid CategoryCreateRequest request) {
-		categoryService.createCategory(request);
+	@GetMapping("/category/{id}")
+	SuccessResponse<CategoryResponse> findById(@PathVariable @Positive Long id) {
+		Category category = categoryService.getCategory(id);
+		CategoryResponse categoryResponse = categoryToResponseMapper.map(category);
+		return SuccessResponse.success(categoryResponse);
+	}
+
+	@PostMapping("/category")
+	SuccessResponse<Void> create(@RequestBody @Valid CategoryCreateRequest request) {
+		categoryService.create(request);
+		return SuccessResponse.success();
+	}
+
+	@PutMapping("/category/{id}")
+	SuccessResponse<Void> update(@PathVariable @Positive Long id, @RequestBody @Valid CategoryUpdateRequest request) {
+		categoryService.update(id, request);
+		return SuccessResponse.success();
+	}
+
+	@DeleteMapping("/category/{id}")
+	SuccessResponse<Void> delete(@PathVariable @Positive Long id) {
+		categoryService.delete(id);
 		return SuccessResponse.success();
 	}
 
```

Comment:
Endpointlere herhangi bir yetki tanımlaması yapılmamış. Her endpoint için bir yetki oluşturup, hem endpointlere hem de rollere bağlayabilir miyiz?

Ek olarak `@RequestMapping` kısmında `/category` kalmış, muhtemelen bu şekilde istek atamayız